public/catalog/kitchensink.yaml

```diff
@@ -104,4 +106,32 @@ nodes:
   label: Node 3
   description: Description 3
   category: grouptests
+- title: Node to test pruning of defaultstest
+  id: defaultprune
+  label: Default prune?
+  category: defaultstest
+  uiSchema:
+    prop4:
+      items:
+        ui:widget: radio
+  schema:
+    type: object
+    additionalProperties: false
+    properties:
+      prop1:
+        title: String with no default
+        type: string
+      prop2:
+        title: String with default empty string
+        type: string
+        default: ''
+      prop3:
+        title: String with default x string
+        type: string
+        default: 'x'
+      prop4:
+        type: array
+        items:
+          type: boolean
+          default: false
```

Comment:
If I add one or more array items, press submit, then remove all of them again, press submit again,

all array items are removed, but the empty array still shows in the toml. Not a problem, but also not sure if needed/intended.